scripts/turtles-quickstart.sh

```diff
@@ -0,0 +1,297 @@
+helm repo update
+helm install rancher-turtles turtles/rancher-turtles --version v0.16.0 \
+    -n rancher-turtles-system \
+    --set rancherTurtles.managerArguments={--insecure-skip-verify} \
+    --set turtlesUI.enabled=true \
+    --dependency-update \
+    --create-namespace --wait \
+    --timeout 180s
+
+show_step "Waiting for deployment capi-controller-manager to be created..."
+
+until kubectl get deployment capi-controller-manager -n capi-system > /dev/null 2>&1; do
+  sleep 1
+done
+kubectl rollout status deployment capi-controller-manager -n capi-system --timeout=180s
+
+# Enable the experimental Cluster topology feature.
+export CLUSTER_TOPOLOGY=true
+
+show_step "Install CAPI Docker provider on ${RANCHER_CLUSTER_NAME}"
+
+clusterctl init --infrastructure docker
+
+# The list of service CIDR, default ["10.128.0.0/12"]
+export SERVICE_CIDR=["10.96.0.0/12"]
+
+# The list of pod CIDR, default ["192.168.0.0/16"]
+export POD_CIDR=["192.168.0.0/16"]
+
+kubectl rollout status deployment capd-controller-manager -n capd-system --timeout=180s
+
+show_step "Create a new example cluster using CAPI Docker provider"
+
+kubectl create namespace capi-clusters
+clusterctl generate cluster ${CAPI_CLUSTER_NAME} --flavor development \
+    --kubernetes-version v1.30.0 \
+    --control-plane-machine-count=1 \
+    --worker-machine-count=1 \
+    -n capi-clusters \
+    > ${CAPI_CLUSTER_NAME}.yaml
+
+kubectl apply -f ${CAPI_CLUSTER_NAME}.yaml
+
+show_step "Waiting for the CAPI cluster to be provisioned ..."
+
+SECONDS=0
+TIMEOUT=240
+until [[ $(kubectl get -n capi-clusters cluster ${CAPI_CLUSTER_NAME} -o jsonpath='{.status.phase}' 2>/dev/null) == "Provisioned" ]]; do
+    [[ $SECONDS -ge $TIMEOUT ]] && echo "timeout reached" && break
+    echo "Waiting for the cluster to be provisioned... ${SECONDS}s"
+    sleep 1 
+done
+echo "Cluster ${CAPI_CLUSTER_NAME} was PROVISIONED by CAPI!"
+
+show_step "Get the kubeconfig for the new cluster"
+
+echo "clusterctl get kubeconfig -n capi-clusters ${CAPI_CLUSTER_NAME} > ${CAPI_CLUSTER_NAME}.kubeconfig"
+until kind get kubeconfig -n ${CAPI_CLUSTER_NAME} &> /dev/null 2>&1; do
+  sleep 1
+done
+clusterctl get kubeconfig -n capi-clusters ${CAPI_CLUSTER_NAME} > ${CAPI_CLUSTER_NAME}.kubeconfig
+
+show_step "Deploying Calico CNI on the new example CAPI cluster"
+
+kubectl --kubeconfig=./${CAPI_CLUSTER_NAME}.kubeconfig \
+    apply -f https://raw.githubusercontent.com/projectcalico/calico/v3.26.1/manifests/calico.yaml
+
+kubectl label namespace capi-clusters cluster-api.cattle.io/rancher-auto-import=true
+
+NODE_IP=$(kubectl get nodes -o jsonpath='{ $.items[*].status.addresses[?(@.type=="InternalIP")].address }')
+
+echo
+echo "SUCCESS: Rancher is ready at:"
+echo "        https://$RANCHER_HOSTNAME"
+echo
+echo "Login: admin"
+echo "Password: ${RANCHER_PASSWORD}"
+echo
+echo "NOTES: To get access to Rancher pleaese add to /etc/hosts a new line:"
+echo "$NODE_IP rancher.loc"
```

Comment:
We need the option to use ngrok, to allow people to use other providers, and to make it work on Mac where Docker runs in a separate network

Reply:
it also prints the default value for rancher hostname

Reply:
done